api/discovery/v1alpha1/foreigncluster_types.go

```diff
@@ -42,12 +42,17 @@ type ForeignClusterSpec struct {
 	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
 	// Important: Run "make" to regenerate code after modifying this file
 
-	ClusterID        string        `json:"clusterID"`
-	Namespace        string        `json:"namespace"`
-	Join             bool          `json:"join"`
-	ApiUrl           string        `json:"apiUrl"`
-	DiscoveryType    DiscoveryType `json:"discoveryType"`
-	AllowUntrustedCA bool          `json:"allowUntrustedCA"`
+	ClusterIdentity  *ClusterIdentity `json:"clusterIdentity"`
```

Comment:
Why this change? 
`ClusterIdentity`  looks weird; `ClusterID` looks much more common (if it is an ID, of course).